presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TransformSpatialPredicates.java

```
@@ -331,9 +399,80 @@ else if (alignment < 0) {
+                .filter(handle -> !metadata.getColumnMetadata(session, tableHandle, handle).isHidden())
+                .collect(toImmutableList());
+        if (visibleColumnHandles.size() != 1) {
+            throw new IllegalArgumentException(format("Expected single column for table %s, but found %s columns", name, columnHandles.size()));
+        }
+
+        ColumnHandle kdbTreeColumn = Iterables.getOnlyElement(visibleColumnHandles);
+
+        List<TableLayoutResult> layouts = metadata.getLayouts(session, tableHandle, Constraint.alwaysTrue(), Optional.of(ImmutableSet.of(kdbTreeColumn)));
+        verify(!layouts.isEmpty(), "Got no layout for table %s", name);
+
+        Optional<String> kdbTree = Optional.empty();
+        try (SplitSource splitSource = splitManager.getSplits(session, layouts.get(0).getLayout().getHandle(), UNGROUPED_SCHEDULING)) {
+            while (!Thread.currentThread().isInterrupted()) {
+                SplitBatch splitBatch = getFutureValue(splitSource.getNextBatch(NOT_PARTITIONED, Lifespan.taskWide(), 1000));
+                List<Split> splits = splitBatch.getSplits();
+
+                for (Split split : splits) {
+                    try (ConnectorPageSource pageSource = pageSourceManager.createPageSource(session, split, ImmutableList.of(kdbTreeColumn))) {
+                        do {
+                            getFutureValue(pageSource.isBlocked());
+                            Page page = pageSource.getNextPage();
+                            if (page != null) {
+                                verify(!kdbTree.isPresent(), "Expected exactly one row for table %s, but found more", name);
+                                verify(page.getPositionCount() == 1, "Expected exactly one row for table %s, but found %s rows", name, page.getPositionCount());
+                                kdbTree = Optional.of(VARCHAR.getSlice(page.getBlock(0), 0).toStringUtf8());
```

Try to parse KDB tree here. Fail fast.